### ISSUE TITLE
CLB: error 503 with TASKS reports

### ISSUE THREAD

**Issue by yroskov** [CatalogueOfLife/backend#1182]
I cannot open TASK reports even with 399 names (50 per page). Screen with endlessly spinning circle. (reported 2022-11-15 via SLAC.

Then, message "Request failed with status code 503" appears.

![image](https://user-images.githubusercontent.com/28574252/202205854-d23cd9b1-3a0d-4a71-abe0-1b922539b571.png)

https://www.checklistbank.org/catalogue/3/dataset/1204/duplicates?acceptedDifferent=true&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=500&minSize=2&mode=STRICT&offset=0&status=accepted&status=synonym&withDecision=false

**Comment by yroskov** [comment]
Just in case, if it's not a server problem. Console says:
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. [duplicates](https://www.checklistbank.org/catalogue/3/dataset/1204/duplicates?_colCheck=c5)
false 2 [index.js:124:22](https://www.checklistbank.org/static/js/components/LayoutNew/index.js)

![image](https://user-images.githubusercontent.com/28574252/202212941-efa16564-e038-4de6-ad16-f4bb4a3c4640.png)

**Comment by yroskov** [comment]
Hmm, TSAKS work well with 990 duplicates per page (= 1167 names) for World Plants: 

https://www.checklistbank.org/catalogue/3/dataset/1141/duplicates?acceptedDifferent=true&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=990&minSize=2&mode=STRICT&offset=0&status=accepted&status=synonym&withDecision=false

Could it be Staphbase data specific problem? @gdower, it's a new version from TW. Perhaps, something new in data may affect TASKS reports

**Comment by yroskov** [comment]
_Another experiment_, SF Orthoptera dataset imported from TaxonWorks: 
TASKS report with 1069 names on the page opens quickly: https://www.checklistbank.org/catalogue/3/dataset/1021/duplicates?acceptedDifferent=false&authorshipDifferent=false&catalogueKey=3&category=binomial&limit=990&minSize=2&mode=STRICT&offset=0&status=accepted&status=synonym&withDecision=true

@gdower & @mdoering, looks like the StaphBase case is not cased by routine export from TaxonWorks. More likely, it cased by changes in StaphBase data.

**Comment by yroskov** [comment]
The same problem appears with newly imported today Systema Dipterorum: 

https://www.checklistbank.org/catalogue/3/dataset/1101/duplicates?acceptedDifferent=false&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=500&minSize=2&mode=STRICT&offset=0&status=synonym&withDecision=false

**Comment by yroskov** [comment]
Two (only two) TASKS reports failed to open in newly imported 3i Auchenorrhyncha:

https://www.checklistbank.org/catalogue/3/dataset/2317/duplicates?acceptedDifferent=false&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=500&minSize=2&mode=STRICT&offset=0&status=synonym&withDecision=false

and 

https://www.checklistbank.org/catalogue/3/dataset/2317/duplicates?catalogueKey=3&category=uninomial&limit=500&minSize=2&mode=STRICT&offset=0&rank=genus&withDecision=false

**Comment by mdoering** [comment]
the 503 error is just our front cache varnish responding that no response could be fetched in an appropriate time. A backend timeout if you like. The actual query here runs far too long to ever finish. The ultimate cause is decorating all records with a full classification. If there are 1000 duplicates looking up 1000 classifications can take some time if the dataset uses strongly nested hierarchies. But Staphbase does not look very deeply nested really. 

https://github.com/CatalogueOfLife/backend/issues/1122 would solve the problem like many other open performance issues.

**Comment by yroskov** [comment]
But why this problem appears only now? Which recent change caused it?

**Comment by mdoering** [comment]
A surprise for me too. Nothing has changed in that area for a long time

**Comment by yroskov** [comment]
2022-12-02: Now all Task reports can be open (checked with StaphBase, 3i & WCVP

**Comment by yroskov** [comment]
Today, 2022-12-14 the problem reported above re-appeared again.

For example, I am not able to open TASKS reports 
ACC-SYN species (different accepted, different authors) 1 of 399
https://www.checklistbank.org/catalogue/3/dataset/1204/duplicates?acceptedDifferent=true&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=500&minSize=2&mode=STRICT&offset=0&status=accepted&status=synonym&withDecision=false
SYN-SYN species (different accepted, different authors) 0 of 972 
https://www.checklistbank.org/catalogue/3/dataset/1204/duplicates?acceptedDifferent=true&authorshipDifferent=true&catalogueKey=3&category=binomial&limit=500&minSize=2&mode=STRICT&offset=0&status=synonym&withDecision=false

After long wait 503 error appear on the screen:
![image](https://user-images.githubusercontent.com/28574252/207637106-28d52da8-44c1-447f-9c17-7d25705a5f20.png)

**Comment by mdoering** [comment]
again Staphbase?

**Comment by yroskov** [comment]
Yes, StaphBase. Let me check WCVP...

**Comment by yroskov** [comment]
Tasks in WCVP works well. Even _ACC-SYN species (different accepted, different authors) 1 of 9016_ with 990 duplicates per page setting.

So, yes, the problem is with StaphBase (exported from TaxonWorks).

Now checking other TW checklists....

**Comment by mdoering** [comment]
Thats interesting. Hope I find some time to look closer what might be the cause

**Comment by yroskov** [comment]
@mdoering, Tasks reports are very slow again (ITIS, 2023-01-04). Even _ACC-ACC species (different authors) of 136 spp_ took few minutes to be opened.

**Comment by yroskov** [comment]
Yes, actual report pages

**Comment by yroskov** [comment]
The problem was with ACC-SYN species (different accepted, different authors) 1023 of 1152 and SYN-SYN species (different accepted, different authors) 1831 of 1973 (error 503, even with the setting "50 duplicates per page").

Today both reports work well.

**Comment by mdoering** [comment]
ok. It looks like it is a complex situation with other concurrent processes, so it's hard to reproduce.
Can you leave a note here each time you encounter it again with exact time (in UTC if possible), and URL being called? Then we can check logs what was happening and try to correlate things.

**Comment by yroskov** [comment]
2023-03-08: working with TASKS in World Ferns, I cannot open report on [ACC-SYN species (different accepted, different authors) 70 of 369](https://www.checklistbank.org/catalogue/3/dataset/1140/duplicates?_colCheck=c5): 

After some time of spinning circle, I got 503 error report:

![image](https://user-images.githubusercontent.com/28574252/223836052-4e3d11c5-7411-478a-a48e-9d1d35158e1e.png)

**Comment by mdoering** [comment]
Was this during the release on the 9th? That might be troublesome to do other things on the project while it is running.

**Comment by yroskov** [comment]
>Was this during the release on the 9th?

No, this was before the release, during my work on duplicates in WFerns

**Comment by mdoering** [comment]
but not during the import, or could that be? It was imported on the 8th: 
> March 8th, 3:54 pm | March 8th, 4:10 pm


If you see this again please log the exact UTC (or local) time

**Comment by yroskov** [comment]
I am logging issue in the GitHub at the time when it was happened. If you can find the time in GitHub, you'll find exact time of last event (well, +1-2 minutes for the screenshot and text).

**Comment by yroskov** [comment]
One thought, if imports (or other running processes) would slow down reports on duplicates, why WFerns were affected, but WPlants were not at very same time?

**Comment by mdoering** [comment]
Because imports only effects queries at specific stages of the import when they both access the same tables. That is a rather rare coincidence, but happens. We could block all access to datasets while an import is running on the database, but that cause more problems than it solves?

**Comment by mdoering** [comment]
<img width="393" alt="image" src="https://user-images.githubusercontent.com/327505/224357943-06a66005-2645-4163-b21f-e07333294b51.png">
so that was 8th march, 9:06pm GMT+1 = CET. Which is 8:06pm UTC if I am right. 4 hours after the import. Not sure if that really collided
